### PR TITLE
feat(ui/ingest): ingestion form for Okta and AzureAD

### PR DESCRIPTION
**Pull request by gaurav2733**
…a and AzureAD

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/azure.ts
Comment (chriscollins3456) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/azure.ts:3]: it looks like this function was copied from `csv.ts` - let's bring that function to a new shared `utils.ts` file and import it in the three recipes instead of copying it twice.

Then, it would be nice to have unit tests on that utility function

**Review by chriscollins3456 — APPROVED** [review]
going to approve assuming that the fields we have here are what ingestion team would recommend but the frontend implementation looks good!

the main change before merging is to bring the copied function into a shared utility file. I'm approving anyways because I don't want to block once that change is made.

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/azure.ts
Comment (gaurav2733) [reply datahub-web-react/src/app/ingest/source/builder/RecipeForm/azure.ts]: updated!

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/azure.ts
Comment (anshbansal) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/azure.ts:4]: Add option for stateful ingestion in advanced section

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/okta.ts
Comment (gaurav2733) [reply datahub-web-react/src/app/ingest/source/builder/RecipeForm/okta.ts:32]: "Can I change source.config.okta_profile_to_username_attr to source.config.email, and is it only the placeholder that needs to be changed? Additionally, should I move the okta_profile_to_username_regex in advance section or filter section?

**Pull request merged**
